src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -284,62 +340,102 @@ boolean canRepair(RepairSegment segment, String keyspace, JmxProxy coordinator,
       repairRunner.updateLastEvent(msg);
       return false;
     }
+    
+    boolean gotMetricsForAllHosts = true;
 
-    for (String hostName : allHosts) {
-      LOG.debug("checking host '{}' for pending compactions and other repairs (can repair?)"
-                + " Run id '{}'", hostName, segment.getRunId());
-      try (JmxProxy hostProxy = context.jmxConnectionFactory.connect(hostName)) {
-        int pendingCompactions = hostProxy.getPendingCompactions();
-        if (pendingCompactions > MAX_PENDING_COMPACTIONS) {
-          LOG.info("SegmentRunner declined to repair segment {} because of too many pending "
-                   + "compactions (> {}) on host \"{}\"", segmentId, MAX_PENDING_COMPACTIONS,
-              hostProxy.getHost());
-          String msg = String.format("Postponed due to pending compactions (%d)",
-              pendingCompactions);
-          repairRunner.updateLastEvent(msg);
-          return false;
+    
+    List<Callable<Optional<HostMetrics>>> getMetricsTasks = allHosts.stream()
+        .map(host -> getMetricsForHost(host))
+        .collect(Collectors.toList());
+    
+    List<Future<Optional<HostMetrics>>> nodesMetrics = Lists.newArrayList();
+
+    try {
+      nodesMetrics = this.metricsGrabberExecutor.invokeAll(getMetricsTasks);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this instance-reference to a static reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)